### PR TITLE
feat: suggest fuzzy + limit as a param

### PR DESCRIPTION
**Pull request by alexgarel**
fixes: https://github.com/openfoodfacts/openfoodfacts-dart/issues/465

Also add fuzzy searching.

### CODE REVIEW

**Comment by alexgarel** [comment]
It would be cool to add tests, but I don't know if I have the time because it will require making a sub to be able to test with  a fake taxonomy.

**Review by monsieurtanuki — APPROVED** [review]
Thank you @alexgarel!
If I sum up, now there's
* the fuzzy search
* a new optional `limit` parameter (default is `25`, max is `400`)

File: cgi/suggest.pl [hunk 1]
```diff
@@ -42,84 +42,146 @@
...
+			# matching at start
+			if ($tagid =~ /^$stringid/) {
 				push @suggestions, $tag;
+				$suggestions_count++;
 			}
-			else {
+			# matching inside
+			elsif ($tagid =~ /$stringid/) {
 				push @suggestions_c, $tag;
+				$suggestions_count++;
```
Comment (stephanegigandet) [inline cgi/suggest.pl:162]: One issue with doing that (which was already present before) is that we will hit the limit with entries that don't start with the searched string, even though we may have some.

e.g. just searching with one letter:

image

So we could create a separate counter for suggestions matching inside instead.

**Comment by alexgarel** [comment]
PR updated to only count for "start with" match to limit traversal.

**Review by stephanegigandet — APPROVED** [review]

**Pull request merged**